lib/class-wp-json-comments-controller.php

```diff
@@ -0,0 +1,470 @@
+		}
+
+		$prepared_args = $this->prepare_item_for_update( $request );
+
+		if ( empty( $prepared_args ) && isset( $request['status'] ) ) {
+			// only the comment status is being changed.
+			$change = $this->handle_status_change( $request['status'], $comment );
+			if ( ! $change ) {
+				return new WP_Error( 'json_comment_failed_edit', __( 'Updating comment status failed.' ), array( 'status' => 500 ) );
+			}
+		} else {
+			$prepared_args['comment_ID'] = $id;
+
+			$updated = wp_update_comment( $prepared_args );
+			if ( 0 === $updated ) {
+				return new WP_Error( 'json_comment_failed_edit', __( 'Updating comment failed.' ), array( 'status' => 500 ) );
+			}
+
+			if ( isset( $request['status'] ) ) {
+				$this->handle_status_change( $request['status'], $comment );
+			}
+		}
+
+		$response = $this->get_item( array(
+			'id'      => $id,
+			'context' => 'edit',
+		) );
+		$response = json_ensure_response( $response );
+		$response->set_status( 201 );
+		$response->header( 'Location', json_url( '/wp/comments/' . $comment->comment_ID ) );
+
+		return $response;
+	}
+
+	/**
+	 * Delete a comment.
+	 *
+	 * @param  WP_JSON_Request $request Full details about the request.
+	 * @return WP_Error|array|mixed
+	 */
+	public function delete_item( $request ) {
+		$id = (int) $request['id'];
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
+
+		$comment = get_comment( $id );
+		if ( empty( $comment ) ) {
+			return new WP_Error( 'json_comment_invalid_id', __( 'Invalid comment ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! $this->check_edit_permission( $comment ) ) {
+			return new WP_Error( 'json_user_cannot_delete_comment', __( 'Sorry, you are not allowed to delete this comment.' ), array( 'status' => 401 ) );
+		}
+
+		$result = wp_delete_comment( $comment->comment_ID, $force );
+		if ( ! $result ) {
+			return new WP_Error( 'json_cannot_delete', __( 'The comment cannot be deleted.' ), array( 'status' => 500 ) );
+		}
+
+		if ( $force ) {
+			return array( 'message' => __( 'Permanently deleted comment' ) );
+		}
+
+		return array( 'message' => __( 'Deleted comment' ) );
+	}
+
+	/**
+	 * Prepare a single comment output for response.
+	 *
+	 * @param  object          $comment Comment object.
+	 * @param  WP_JSON_Request $request Request object.
+	 * @return array $fields
+	 */
+	public function prepare_item_for_response( $comment, $request ) {
+		$fields = array(
+			'id'           => (int) $comment->comment_ID,
+			'post_id'      => (int) $comment->comment_post_ID,
+			'parent_id'    => (int) $comment->comment_parent,
+			'user_id'      => (int) $comment->user_id,
+			'author'       => $comment->comment_author,
+			'author_email' => $comment->comment_author_email,
+			'author_url'   => $comment->comment_author_url,
+			'date'         => json_mysql_to_rfc3339( $comment->comment_date ),
```

Comment:
@danielbachhuber comments don't have modified_date like posts.  We could make it `comment_date`?  `publish_date` seems confusing since you can modify the date.